nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/P2Attributes.java

```diff
@@ -94,9 +94,21 @@ public Builder extension(final String extension) {
     }
 
     public P2Attributes build() {
+      if (isNull(pluginName, componentName, componentVersion, path, fileName, extension, null)) {
+        return null;
+      }
       return new P2Attributes(this);
     }
```

Comment:
isNull(pluginName, componentName, componentVersion, path, fileName, extension, null) 
why the last value is null?
why if all values are null the function returns null?
Its not a obvious behavior for build()

Reply:
last null is typo, fixed, if all values in object are not assign - there is no sense in object creation.